### DISCUSSION TITLE
File type reverse engineering thread

### DISCUSSION THREAD

**Discussion by applecuckoo** [lfhacks/Virtuos-LeapFrog-game-tools#5] [Ideas]
Hi all! Just putting all my findings here from TFM so far:

## Standard file formats:

- Ogg Vorbis (.ogg) - general sound
- IMA ADPCM encoded WAV files (.wav) - general sound
- XML (.xml) - markup/general config
- Lua 5.1 bytecode (.luac) - scripts

## Proprietary file formats:

- .tal - unknown, but related to textures
- .abf - unknown, but related to animation
- .udi (User Defined Image) - textures
- .waf - archive

I'll probably update this when I've got more time...

**Comment by applecuckoo** [comment]
@BLiNXthetimesweeperGOD Looks like you might've missed this? I saw your update to UDI.md and checked a few of the UDIs from TFM. It looks like some of those have a non-zero value at bytes 0x15-17, so it seems like only byte 0x14 is the compresssion flag.

Also, what's currently marked as the image offset (bytes 0x08-0x09) isn't always 36, so that should also be taken out - maybe it represents something else? I'd have to do some decompiling to figure that out, but I don't really know any ARM assembly skills, so I can only stare at the binary in Ghidra and hope it makes sense...

Lastly, this is just a thought, but *maybe* we could expand the scope of this project to include the other Explorer devs? I've already made some progress on deciphering Other Ocean Interactive's .oot textures and am somewhat making leeway on InLight and 1st Playable Productions' formats. I even discovered a handful of Nintendo DS-related strings in 1st Playable's fs.arc files which was odd, but plausible - apparently they've worked on a few DS titles in the past. As a start, I'll probably put them in my personal LeapNotes repo once I get the time to do so.

Anyways, that's all my ramble for now - I'll see if I manage to write some documentation. If I do, I'll update this comment.

**Comment by BLiNXthetimesweeperGOD** [comment]
I could start a repository for general LeapFrog research. It'd give me an excuse to put my incomplete WayForward .ANM sprite format research somewhere.

**Comment by BLiNXthetimesweeperGOD** [comment]
It's up! Feel free to contribute anything you want to it. You can even edit the main readme if you want (at least, I hope you can - I'm still kind of new to actually using Github and don't know how everything works yet).

https://github.com/lfhacks/General-LeapFrog-game-research
